common/char_difference.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package com.google.startup.common;
+
+option java_package = "com.google.startup.common";
+option java_multiple_files = true;
+
+// Holds information about a character difference.
+message CharDifference {
+  // The index in which the difference has occurred. For example, if we have
+  // the two strings, "Exampl" and "Example", the index would be 6.
+  int64 index = 1;
+
+  // The string of the difference. Limited to hold only one character. In the
+  // example of the two strings "Exampl" and "Example" it would be the last "e".
+  string difference = 2;
+
+  // The type of the difference, i.e. what change has occurred. In the example
+  // of the two strings "Exampl" and "Example" it would be ADDITION.
+  DifferenceType type = 3;
+}
+
+/** Describes the type of the difference. */
+enum DifferenceType {
+  // No change has occurred. The source is equal to the
+  // destination.
+  NO_CHANGE = 0;
+
+  // An addition of new characters has occurred.
+  ADDITION = 1;
+
+  // A deletion of characters has occurred. The source is missing several
```

Comment:
If CharDifference.difference is only 1 character, why is this plural? "characters", "missing several" (also in ADDITION).